File: src/lib/api/projects.ts
import { apiFetch } from "./client";

export interface ProjectPayload {
  name: string;
  client: string;
  startDate: string;
  endDate?: string | null;
  status: "active" | "completed" | "on-hold";
  description?: string;
}

export interface ProjectResponse {
  id: string;
  name: string;
  client: string;
  client_id: string;
  startDate: string;
  endDate?: string | null;
  status: "active" | "completed" | "on-hold";
  description?: string;
}

export async function fetchProjects(): Promise<ProjectResponse[]> {
  return apiFetch<ProjectResponse[]>("/projects");
}

export async function createProject(payload: ProjectPayload): Promise<ProjectResponse> {
  return apiFetch<ProjectResponse>("/projects", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export async function updateProject(id: string, payload: Partial<ProjectPayload>): Promise<ProjectResponse> {
  return apiFetch<ProjectResponse>(`/projects/${id}`, {
    method: "PUT",
    body: JSON.stringify(payload),
  });
}

export async function deleteProject(id: string): Promise<{ status: string; message: string }> {
  return apiFetch<{ status: string; message: string }>(`/projects/${id}`, {
    method: "DELETE",
  });
}

export async function generateProjectTasks(id: string): Promise<{ status: string; message: string }> {
  return apiFetch<{ status: string; message: string }>(`/projects/${id}/generate-tasks`, {
    method: "POST",
  });
}

export async function fetchProjectTasks(id: string): Promise<any[]> {
  return apiFetch<any[]>(`/projects/${id}/tasks`);
}

File: backend/app/schemas/projects.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, Field

class ProjectCreateRequest(BaseModel):
    name: str = Field(..., max_length=150)
    client: str = Field(..., description="Client name")
    start_date: str = Field(..., alias="startDate")
    end_date: Optional[str] = Field(None, alias="endDate")
    status: str = Field("active", description="active, completed, or on-hold")
    description: Optional[str] = None

    class Config:
        populate_by_name = True

class ProjectUpdateRequest(BaseModel):
    name: Optional[str] = Field(None, max_length=150)
    client: Optional[str] = Field(None, description="Client name")
    start_date: Optional[str] = Field(None, alias="startDate")
    end_date: Optional[str] = Field(None, alias="endDate")
    status: Optional[str] = Field(None, description="active, completed, or on-hold")
    description: Optional[str] = None

    class Config:
        populate_by_name = True

class ProjectResponse(BaseModel):
    id: UUID
    name: str
    client: str  # Client name
    client_id: UUID
    start_date: str = Field(..., alias="startDate")
    end_date: Optional[str] = Field(None, alias="endDate")
    status: str
    description: Optional[str] = None

    class Config:
        from_attributes = True
        populate_by_name = True
